src/_includes/layouts/base.njk

```diff
@@ -4,6 +4,28 @@
     <meta charset="utf-8">
     <title>{% if '/' === page.url %}V8 JavaScript engine{% else %}{{ (renderData.title or title) | markdown | striptags | decodeHtmlEntities }} · V8{% endif %}</title>
     <meta name="viewport" content="width=device-width,initial-scale=1">
+    <meta name="color-scheme" content="dark light">
+    <script>
+      // If `prefers-color-scheme` is not supported, fall back to light mode.
+      // In this case, light.css will be downloaded with `highest` priority.
+      if (window.matchMedia('(prefers-color-scheme: dark)').media === 'not all') {
+        document.documentElement.style.display = 'none';
+        document.head.insertAdjacentHTML(
+            'beforeend',
+            '<link rel="stylesheet" href="/_css/light.css" onload="document.documentElement.style.display = ``">'
+        );
+      }
+    </script>
+    <!--
+      Conditionally either load the light or the dark stylesheet. The matching file
+      will be downloaded with `highest`, the non-matching file with `lowest`
+      priority. If the browser doesn't support `prefers-color-scheme`, the media
+      query is unknown and the files are downloaded with `lowest` priority (but
+      above I already force `highest` priority for my default light experience).
+    -->
+    <link rel="stylesheet" href="/_css/dark.css" media="(prefers-color-scheme: dark)">
+    <link rel="stylesheet" href="/_css/light.css" media="(prefers-color-scheme: no-preference), (prefers-color-scheme: light)">
+    <!-- The main stylesheet -->
```

Comment:
I think in this case we can avoid the complexity by inlining the contents of `light.css` and `dark.css` into `main.css`.

Reply:
The only change to `base.jk` is then the addition of `<meta name="color-scheme" content="dark light">`. WDYT?

Reply:
This makes sure we really only load what we actually need, and with HTTP/2 small files aren't a problem in this case. It also allows the toggle to just disable or enable the override without the need for an extra "dark" class or something.

Reply:
> This makes sure we really only load what we actually need, and with HTTP/2 small files aren't a problem in this case.

I’m more worried about the additional content-blocking `<script>` on every page than about loading a few CSS bytes that we might not need.

> It also allows the toggle to just disable or enable the override without the need for an extra "dark" class or something.

I don’t understand this part. With what I had in mind, there is no `dark` class. We’d just add the contents of `light.css` to the top of `main.css`, and then add:

```css
@media (prefers-color-scheme: dark) {
  /* contents of dark.css go here */
}
```

Am I misunderstanding?

Reply:
Another benefit is that this way, everything would Just Work™ with our service worker, without us needing to update our Workbox configuration.

Reply:
It's explained in the [article](https://web.dev/prefers-color-scheme#loading-strategy), but the *tl;dr* is: [not all browsers](https://caniuse.com/#feat=prefers-color-scheme) understand the `prefers-color-scheme` media query, which leads to such browsers downloading stylesheets with unknown media query with [lowest priority](https://blog.tomayac.com/2018/11/08/why-browsers-download-stylesheets-with-non-matching-media-queries-180513/). This is bad, because either `light.css` or `dark.css` are needed, apart from `main.css`. The small [inline script](https://github.com/v8/v8.dev/blob/754f5df29b74f856bd0698b78e04e7ed538eb85a/src/_includes/layouts/base.njk#L8-L18) makes sure this doesn't happen and `light.css`(as an arbitrarily chosen default) is loaded with highest priority.

The way the toggle makes sure even browsers that don't support the media query get the theme switching experience is by changing the media to something all browsers know how to deal with:

<img width="497" alt="Screen Shot 2019-10-03 at 18 49 21" src="https://user-images.githubusercontent.com/145676/66147173-94175b00-e60e-11e9-85c0-b7d908b745fc.png">

You can see this in action on [this old Chrome 74 browser session](https://www.browserling.com/browse/win/7/chrome/74/https%3A%2F%2Fdark-mode-baseline.glitch.me%2F), only Chrome 76 started supporting `prefers-color-scheme`.

Reply:
So `<dark-mode-toggle>` *requires* separate stylesheets?

Reply:
It absolutely [doesn't](https://github.com/GoogleChromeLabs/dark-mode-toggle#-using-a-css-class-that-you-toggle), but I'd still argue it's the nicer architecture, especially once you start working with bigger apps. For example, *"[i]mplementing Dark Mode [in WebKit's Web Inspector] took over **1,000 lines of CSS**"* ([source](https://webkit.org/blog/8892/dark-mode-in-web-inspector/)).

Reply:
Great to hear! I completely agree that for more complex apps having separate stylesheets is the better approach.

However, for the extremely minimal/simple case of v8.dev, I **strongly** prefer the class-based approach as it is a much less invasive change. We don’t have to change our HTML, we don’t have to change our service worker config, and we don’t have to change our build pipeline.

I look at it this way: the current patch adds 667 bytes to each and every HTML page on v8.dev to try and prevent loading ~700 bytes of CSS (either `light.css` or `dark.css`) unnecessarily. You’ve done such an amazing job keeping both `light.css` and `dark.css` super small — let’s leverage it!

Reply:
@mathiasbynens Commit https://github.com/v8/v8.dev/pull/254/commits/e82bb820e649c847670cdac171ebf4c56e84806b changes the logic to now use a class `dark` with one CSS file instead of conditional loading several CSS files. PTAL.